File: Seccion 11 - Funciones en C++/Seccion_11_Dart/20_ejercicio_16.dart

import 'dart:io';

int nfilas, ncols;
List<List> m = new List<List>();

void pedirDatos() {
  int val;

  stdout.writeln('Digite el número de filas: ');
  nfilas = int.parse(stdin.readLineSync());
  stdout.writeln('Digite el número de columnas: ');
  ncols = int.parse(stdin.readLineSync());

  for (int i = 0; i < nfilas; i++) {
    m.insert(i, new List<int>());
    for (int j = 0; j < ncols; j++) {
      stdout.writeln('[$i][$j] Digite un número: ');
      val = int.parse(stdin.readLineSync());
      m[i].insert(j, val);
    }
  }
}

int comprobarMenorElemento() {
  int fila, menor = 99999;

  stdout.writeln('\nDigite un número de fila para comprobar menor elemento: ');
  fila = int.parse(stdin.readLineSync());

  for (int i = 0; i < nfilas; i++) {
    for (int j = 0; j < ncols; j++) {
      if (i == fila) {
        if (m[i][j] < menor) {
          menor = m[i][j];
        } 
      }
    }
  }

  return menor;
}

void main() {

  pedirDatos();

  int menorElemento = comprobarMenorElemento();

  print('\nEl menor elemento es: $menorElemento');
}

File: Seccion 11 - Funciones en C++/Seccion_11_Python/20_ejercicio_16.py

nfilas = 0
ncols = 0
m = []

def pedir_datos():
    global nfilas, ncols, m
    
    nfilas = int(input('Digite el número de filas: '))
    ncols = int(input('Digite el número de columnas: '))
    
    for i in range(nfilas):
        fila = []
        for j in range(ncols):
            val = int(input('[{}][{}] Digite un número: '.format(i, j)))
            fila.append(val)
        m.append(fila)

def comprobar_menor_elemento():
    menor = 99999
    
    fila = int(input('\nDigite el número de fila para comprobar menor elemento: '))
    
    for i in range(nfilas):
        for j in range(ncols):
            if i == fila:
                if m[i][j] < menor:
                    menor = m[i][j]
    
    return menor

pedir_datos()

menor_elemento = comprobar_menor_elemento()

print('\nEl menor elemento es: {}'.format(menor_elemento))